pallets/parachain-staking/src/weights.rs

```diff
@@ -148,10 +148,13 @@ impl<T: frame_system::Config> WeightInfo for SubstrateWeight<T> {
 			.saturating_add(T::DbWeight::get().reads(7 as Weight))
 			.saturating_add(T::DbWeight::get().writes(4 as Weight))
 	}
-	fn execute_leave_candidates() -> Weight {
-		(68_763_000 as Weight)
-			.saturating_add(T::DbWeight::get().reads(10 as Weight))
-			.saturating_add(T::DbWeight::get().writes(7 as Weight))
+	fn execute_leave_candidates(x: u32) -> Weight {
+		(0 as Weight) // Standard Error: 8_000
+			.saturating_add((27_557_000 as Weight).saturating_mul(x as Weight))
+			.saturating_add(T::DbWeight::get().reads(6 as Weight))
+			.saturating_add(T::DbWeight::get().reads((2 as Weight).saturating_mul(x as Weight)))
+			.saturating_add(T::DbWeight::get().writes(3 as Weight))
+			.saturating_add(T::DbWeight::get().writes((2 as Weight).saturating_mul(x as Weight)))
```

Comment:
This change makes it quite expensive to leave if you have a lot of delegators (something you don't directly control), but that seems perfectly reasonable; you're affecting a lot of other people.

One thing we should keep in mind is that there is some point where this is so "heavy" that it can't be executed in one block.

Reply:
This was actually not yet changed in this PR. It was in #1207 , so I do need to rerun this and a few other benchmarks.

Reply:
It became even more expensive